### ISSUE TITLE
flyingpio: ppm is broken

### ISSUE THREAD

**Issue by mlyle** [d-ronin/dRonin#1420]
Failsafes; intermittently flashes correct values according to @mhv-shared from IRC.  Have not tried myself.

**Comment by xfce** [comment]
i change the pios_flyingpio.c's cmd->receiver_protocol = HWSHARED_PORTTYPES_SBUS
to cmd->receiver_protocol = HWSHARED_PORTTYPES_PPM
then no change, still only found the throttle stick in GCS's wizard ,
try both quitoxe and atifice branch

**Comment by jihlein** [comment]
I've been looking into this and here's what I've found so far:
1)Made the changes in #2043 to correct RTC frequency.  Verified the RTC frequency with a Logic Analyzer to be 625 Hz.  PPM still does not work correctly.
2)Made a new time base definition TIM14 with the TIM period = 0xFFFF.  PPM still does not work correctly.
3)Changed the processor clock from 40 MHz to 48 MHz.  PPM still does not work correctly.

Interesting things to note:
1)After the above changes, PPM will sometimes lock, after waiting a while,  This wait has been observed to be as short as 56 seconds, or as long as 203 seconds, so it's not consistent.  And sometimes I've waited 5 plus minutes with no lock.  I'm not sure if all of the changes, some of the changes, or one of the changes above are responsible for this.
2)Once locked, however, I can cycle the RX power off and on with immediate lock recovery, and I can do this multiple times with no failure to immediately lock.
3)The PWM frame time in the code is 20 Msec.  I'm using a D4R-II with a 27 mSec frame time.  Don't know if this matters too much as it works fine on the other target boards I've tried, but it might be worth experimenting with.
4)I don't know if it's somehow related, but now that I'm paying more attention to this again, I'm occasionally seeing a "fpio: VERYBAS: Bad CRC response on data XXXX" report on the RPi console screen.
5)I've tried lowering the SPI clock rate from 15 MHz to 10 MHz with no change to the noted behaviors.
6)FWIW, same behaviors are noted whether or not I have the GPS and external mag connected and running.

To be continued..... :)
